### PR TITLE
Add support for `com.google.protobuf:protobuf-java-util:3.21.12`

### PR DESCRIPTION
**Pull request by linghengqian**
## What does this PR do?

- Fixes https://github.com/oracle/graalvm-reachability-metadata/issues/165 .

## Checklist before merging
- [x] I have properly formatted metadata files (see [CONTRIBUTING](/oracle/graalvm-reachability-metadata/blob/master/CONTRIBUTING.md) document)
- [x] I have added thorough tests. (see [this](/oracle/graalvm-reachability-metadata/blob/master/CONTRIBUTING.md#Tests))

### CODE REVIEW

File: tests/src/com.google.protobuf/protobuf-java-util/3.21.12/src/test/resources/META-INF/native-image/protobuf-java-util-test-metadata/reflect-config.json [hunk 1]
```diff
@@ -0,0 +1,6379 @@
+[
```
Comment (mhalbritter) [inline tests/src/com.google.protobuf/protobuf-java-util/3.21.12/src/test/resources/META-INF/native-image/protobuf-java-util-test-metadata/reflect-config.json:1]: Hey - why is this file necessary?

Comment (linghengqian) [reply tests/src/com.google.protobuf/protobuf-java-util/3.21.12/src/test/resources/META-INF/native-image/protobuf-java-util-test-metadata/reflect-config.json:1]: - This file is all GraalVM metadata needed by the test class itself, I haven't tried to simplify it. According to the unit test I did on the ShardingSphere side, protobuf requires a lot of reflection.

Comment (mhalbritter) [reply tests/src/com.google.protobuf/protobuf-java-util/3.21.12/src/test/resources/META-INF/native-image/protobuf-java-util-test-metadata/reflect-config.json:1]: There are a lot of references to classes in there which are not test-specific, like `"name": "com.google.protobuf.Any"` or `"name": "com.google.protobuf.BoolValue"`. My problem is that this could lead to the test passing even if the metadata in the `metadata/` folder is not complete. Can you try to get rid of that file or at least condense it so that it only references classes from your test?

Comment (linghengqian) [reply tests/src/com.google.protobuf/protobuf-java-util/3.21.12/src/test/resources/META-INF/native-image/protobuf-java-util-test-metadata/reflect-config.json:1]: - I made the following changes in https://github.com/linghengqian/graalvm-trace-metadata-smoketest/commit/f9de2103ebb52208ccb28fa0098a3322d0970a60 and put them into the current PR.
- Replace most of `"typeReachable": "com.google.protobuf.util.JsonFormatTest"` with `"typeReachable": "com.google.protobuf.GeneratedMessageV3"`.
- Remove most entries for `"typeReachable": "com.google.protobuf.util.DurationsTest"` and `"typeReachable": "com.google.protobuf.util.FieldMaskTreeTest"`.
- The reason why the file appears to have many lines is because each entry is very large, making the line number bloated.
- image

**Review by mhalbritter — APPROVED** [review]
LGTM!

**Review by vjovanov — APPROVED** [review]

File: tests/src/com.google.protobuf/protobuf-java-util/3.21.12/src/test/resources/META-INF/native-image/protobuf-java-util-test-metadata/reflect-config.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (linghengqian) [reply tests/src/com.google.protobuf/protobuf-java-util/3.21.12/src/test/resources/META-INF/native-image/protobuf-java-util-test-metadata/reflect-config.json:1]: - This is generated by conditional agent mode from the same `user-code-filter.json`. Since there are some `name` entries that has nothing to do with the dependent library, I stripped it to this file.
